autoscale/src/main/java/com/sequenceiq/periscope/monitor/evaluator/CronTimeEvaluator.java

```diff
@@ -80,15 +108,21 @@ public void execute() {
         long start = System.currentTimeMillis();
         Cluster cluster = clusterService.findById(clusterId);
         MDCBuilder.buildMdcContext(cluster);
-        publishIfNeeded(alertRepository.findAllByCluster(clusterId));
+        publishIfNeeded(alertRepository.findAllByClusterIdOrderById(clusterId));
         LOGGER.debug("Finished cronTimeEvaluator for cluster {} in {} ms", cluster.getStackCrn(), System.currentTimeMillis() - start);
     }
 
-    private void publishIfNeeded(List<TimeAlert> alerts) {
+    protected void publishIfNeeded(List<TimeAlert> alerts) {
+        TimeAlert triggeredAlert = null;
         for (TimeAlert alert : alerts) {
-            if (isPolicyAttached(alert) && isTrigger(alert)) {
+            boolean alertTriggerable = isTrigger(alert);
+            if (isPolicyAttached(alert) && alertTriggerable && null == triggeredAlert) {
                 publish(alert);
-                break;
+                triggeredAlert = alert;
```

Comment:
All of the history changes here are part of another patch right? Can you please backport that independently of this change?

Reply:
done.